File: 78-subsets/78-subsets.py
import itertools
class Solution(object):
    def subsets(self, nums):
        """
        :type nums: List[int]
        :rtype: List[List[int]]
        """
        res = []
        def subsets(i,arr,lis):
            if i>=len(arr):
                l_copy = []
                l_copy.extend(lis)
                res.append(l_copy)
                return
            
            lis.append(arr[i])
            subsets(i+1,arr,lis)
            lis.pop()
            
            subsets(i+1,arr,lis)
        
        subsets(0,nums,[])
        return res

File: 78-subsets/78-subsets.cpp
class Solution {
public:
    vector<vector<int>> res;
    void solve(int ind,vector<int>& nums,vector<int>& ds){
        if(ind==nums.size()){
            res.push_back(ds);
            return;
        }
        
        ds.push_back(nums[ind]);
        solve(ind+1,nums,ds);
        ds.pop_back();
        solve(ind+1,nums,ds);
    }
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<int> ds;
        solve(0,nums,ds);
        return res;
    }
};